scripts/mppc_to_multiprobe.py

```diff
@@ -0,0 +1,298 @@
+from libtiff import TIFF
+from odemis.util import almost_equal
+import sys
+from scipy.ndimage.measurements import center_of_mass
+
+
+def preprocess_images(image):
+    """
+    Pre-processing the data to calculate the corrections. The size of the input image is well defined (6400px x 6400px).
+    The image is acquired with one beamlet (E4/(5,4)) and contains 64 cell images (8x8; 800px x 800px; 30um x 30um).
+    Each cell image shows the corresponding active mppc cell image as a bright square. All other cells are inactive
+    and appear dark in the single cell image.
+    The single cell image covers a field of view larger than all mppc cells.
+    The mppc cell image located in the single cell image (bright square) consists of a fixed size of about
+    3um x 3um (80px x 80px), which corresponds to about 1% of the area of a single cell image.
+    Depending on which of the 64 cell images, the bright square is located in a different position of the image.
+    The 64 cell images are stitched together. A projection of all single cell images onto each other would result
+    in one image with all mppc cell images being bright. As the image is recorded using the E4 beamlet,
+    the E4 cell image should show the mppc cell image at the center of the image if correctly aligned.
+    The corner images A1, A8, H1 and H8 are used to correct for rotation, the E4 image is used to correct for an offset.
+    :param image: (DataArray) Image of size 6400px x 6400px made up by 64 cell images (800px x 800px), each containing
+                the corresponding mppc cell image (bright square) at a given position, which are shifted by the size
+                of the square in respect to each other.
+    :returns: (list of 4 ndarrays) The preprocessed corner images. Shape is (y, x).
+              (ndarray) The E4 cell image. Shape is (y, x).
+    """
+
+    # make sure it is grey scale
+    if len(image.shape) > 2:
+        image = RGB2Greyscale(image)
+
+    # get the 4 corner mppc cell images: A1, H1, A8, H8
+    # shape of full mppc is (8, 8)
+    cell_size_y = int(image.shape[0] / 8)
+    cell_size_x = int(image.shape[1] / 8)
+    image_A1 = image[0:cell_size_y, 0:cell_size_x]
+    image_H1 = image[0:cell_size_y, -cell_size_x:]
+    image_A8 = image[-cell_size_y:, 0:cell_size_x]
+    image_H8 = image[-cell_size_y:, -cell_size_x:]
+    image_E4 = image[cell_size_y*3:cell_size_y*4, cell_size_x*4:cell_size_x*5]
+
+    images = [image_A1, image_A8, image_H1, image_H8, image_E4]
+
+    # thresholding
+    # Known:
+    #   - size of single image (800px x 800px => 30um x 30um = 900 um2)
+    #   - size of active mppc cell image (bright square) approximately the same at all times
+    #     (3um x 3um = 9um2 => 1% of image)
+    # set 99% of the pixels with the lowest values to 0, set the rest of the pixels to 1
+    for num_img, img in enumerate(images):
+        shape = img.shape
+        img_histogram = numpy.bincount(numpy.ravel(img))
```

Comment:
Note that this only works if the dtype is uint8 or uint16. This is probably always the case, but you might want to add a comment about this. Otherwise, you'd need to use some more fancy function. see odemis.util.img.histogram()

BTW, you can just write: `numpy.bincount(img.flat)`.